crates/containerd-shim-wasm/src/sandbox/stdio.rs

```diff
@@ -14,51 +17,73 @@ pub struct Stdio {
 }
 
 impl Stdio {
-    pub fn redirect(&self) -> Result<()> {
+    pub fn redirect(self) -> Result<()> {
         self.stdin.redirect()?;
         self.stdout.redirect()?;
         self.stderr.redirect()?;
         Ok(())
     }
+
+    pub fn take(&self) -> Self {
```

Comment:
do I understand properly that it is this `take()` that transfers the ownership?

Reply:
Yes, this is the `take` called that transfers the ownership.
It retuns a new `Stdio` and leaves behing a... `None`-ish `Stdio`.
Also now doing `redirect()` consumes the object, forcing you to `take()` it first (`clone()` would also work).